### PR TITLE
feat: project upgrade

### PR DESCRIPTION
**Pull request by antomor**
## What

- Upgrade the project dependencies
- Replace web3 with ethers.js
- refactor project structure
- adds useful scripts and removes useless ones
- strengthens linting and formatting

## Why

- Dependencies were outdated
- Reduce technical debt

### CODE REVIEW

**Comment by antomor** [comment]
This PR depends on #258

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        run: npm ci --no-audit
+
+      - name: Check Codestyles errors
+        run: npm run format
+
+      - name: Check Linter bugs for Typescript
+        run: npm run lint
+
+      - name: Run docker image
+        run: docker run -d -p 127.0.0.1:4444:4444 -p 127.0.0.1:4445:4445 --name enveloping-rskj -it -v $PWD/docker/logback.xml:/etc/rsk/logback.xml -v $PWD/docker/node.conf:/etc/rsk/node.conf rsksmart/rskj:FINGERROOT-5.0.0 --regtest
```
Comment (franciscotobar) [inline .github/workflows/main.yml:23]: Why do we need to map the port 4445?

File: Readme.md [hunk 2]
```diff
@@ -6,20 +6,19 @@ A secure transaction relay system to enable users to pay fees using ERC-20 token
...
+RIF Relay takes its inspiration from the [Gas Station Network (GSN) project](https://github.com/opengsn/gsn). GSN is a decentralized system that improves dApp usability without sacrificing security. In a nutshell, GSN abstracts away gas (used to pay transaction fees) to minimize onboarding and UX friction for dApps. With GSN, “gasless clients” can interact with smart contracts paying for gas with tokens instead of native-currency.
 
 Our main objective is to provide the RSK ecosystem with the means to enable blockchain applications and end-users (wallet-apps) to pay for transaction fees using tokens (e.g. RIF tokens), and thereby remove the need to acquire RBTC in advance.
 
 It is important to recall that - as a security measure - the version 1 contracts deployed on Mainnet have limits on the staked amounts to operate, these limits were removed in version 2.
 
 ## Modules
 
-Rif Relay is built in modules, the entire system is made up by 4 modules.
+Rif Relay is built in modules, the entire system is made up by 3 modules.
```
Comment (franciscotobar) [inline Readme.md:17]:
```suggestion
RIF Relay is built in modules, the entire system is made up by 3 modules.
```

File: Readme.md [hunk 3]
```diff
@@ -6,20 +6,19 @@ A secure transaction relay system to enable users to pay fees using ERC-20 token
...
 
-1. [Rif Relay Contracts](https://github.com/anarancio/rif-relay-contracts) contains all the contracts used by the Rif Relay System.
-2. [Rif Relay Common](https://github.com/infuy/rif-relay-common) it’s a library that contains all the common code used by other modules.
-3. [Rif Relay Client](https://github.com/infuy/rif-relay-client) contains a library to interact with the relay server.
-4. [Rif Relay Server](https://github.com/infuy/rif-relay-server) has all the relay server code, you can use it as a library or run the server directly from there.
+1. [Rif Relay Contracts](https://github.com/rsksmart/rif-relay-contracts) contains all the contracts used by the Rif Relay System.
```
Comment (franciscotobar) [inline Readme.md:19]:
```suggestion
1. [RIF Relay Contracts](https://github.com/rsksmart/rif-relay-contracts) contains all the contracts used by the Rif Relay System.
```

File: Readme.md [hunk 4]
```diff
@@ -6,20 +6,19 @@ A secure transaction relay system to enable users to pay fees using ERC-20 token
...
+2. [Rif Relay Client](https://github.com/rsksmart/rif-relay-client) contains a library to interact with the relay server.
```
Comment (franciscotobar) [inline Readme.md:20]:
```suggestion
2. [RIF Relay Client](https://github.com/rsksmart/rif-relay-client) contains a library to interact with the relay server.
```

File: Readme.md [hunk 5]
```diff
@@ -6,20 +6,19 @@ A secure transaction relay system to enable users to pay fees using ERC-20 token
...
+3. [Rif Relay Server](https://github.com/rsksmart/rif-relay-server) has all the relay server code. You can run the server directly from there.
```
Comment (franciscotobar) [inline Readme.md:21]:
```suggestion
3. [RIF Relay Server](https://github.com/rsksmart/rif-relay-server) has all the relay server code. You can run the server directly from there.
```

Comment (franciscotobar) [reply Readme.md:21]: Maybe my point here is to use RIF or Rif.

File: config/local.json5 [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  app: {
+    devMode: true,
+    checkInterval: 10,
+    logLevel: 5,
+    workdir: '/tmp/server'
```
Comment (franciscotobar) [inline config/local.json5:6]:
```suggestion
    workdir: './tmp/server'
```
Or are we planning to use the `tmp` system folder?

File: scripts/GasEstimation.ts [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+
+  const deployRequest = {
+    request: {
+      data: '0x00',
+      from: owner.address,
+      nonce: '0',
+      relayHub: relayHub.address,
+      to: constants.AddressZero,
+      tokenAmount,
+      tokenContract: constants.AddressZero,
```
Comment (franciscotobar) [inline scripts/GasEstimation.ts:362]:
```suggestion
     tokenContract: token.address,
```
This is what we discussed with Raul and needs to be fixed.

File: config/local.json5
Comment (franciscotobar) [inline config/local.json5:1]: I dont think that we need this file :)

**Review by franciscotobar — APPROVED** [review]
LGTM

File: config/local.json5 (hunk 6 above)
Comment (antomor) [reply config/local.json5:6]: yup, because that one is automatically deleted

File: config/local.json5
Comment (antomor) [reply config/local.json5:1]: I may be wrong, but IIRC that file is used for the tests

Comment (franciscotobar) [reply config/local.json5:1]: sorry you are right, we need it for the `GasEstimation`

**Pull request merged**